### PR TITLE
layer_stack - rounding errors

### PR DESCRIPTION
**Pull request by HelgeGehring**
@simbilod could you have a look?

### CODE REVIEW

File: gdsfactory/technology/layer_stack.py [hunk 1]
```diff
@@ -143,7 +143,9 @@ def get_component_with_net_layers(
             for polygon in polygons:
                 # If polygon belongs to port, create a unique new layer, and add the polygon to it
 
-                if gdstk.inside([port.center], gdstk.Polygon(polygon))[0]:
+                if gdstk.inside(
+                    [port.center], gdstk.offset(gdstk.Polygon(polygon), 0.001)
```
Comment (simbilod) [inline gdsfactory/technology/layer_stack.py:147]: Good idea, although we could also use the DBU unit

like here: https://github.com/gdsfactory/gplugins/blob/151feb0e123f927351d32ec91446af8ac93ac5c6/gplugins/gmsh/parse_gds.py#L32

**Comment by joamatab** [comment]
is this ready to merge?

**Comment by HelgeGehring** [comment]
@simbilod I've updated this

**Comment by simbilod** [comment]
this can be merged

**Pull request merged**